hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/virtual/ViewTable.java

```diff
@@ -80,6 +81,12 @@ public RelNode getViewRel() {
         return viewRel;
     }
 
+    @Override
+    public boolean isStreaming() {
+        getFields(); // called for the side effect of calling initFields()
```

Comment:
This whole mechanism is way too complex for what it does imo. Maybe worth a revisit later.